CMakeLists.txt

```diff
@@ -344,34 +399,51 @@ if(BUILD_PYTHON)
     gen_nvfuser_version ALL
     DEPENDS ${NVFUSER_ROOT}/nvfuser/version.py
   )
-  add_dependencies(${NVFUSER} gen_nvfuser_version)
+  add_dependencies(nvf_py_internal gen_nvfuser_version)
 
-  target_compile_options(${NVFUSER} PRIVATE "-DTORCH_CUDA_BUILD_MAIN_LIB")
-
-  # NB: This must be target_compile_definitions, not target_compile_options,
-  # as the latter is not respected by nvcc
-  target_compile_definitions(${NVFUSER} PRIVATE "-DTORCH_CUDA_BUILD_MAIN_LIB")
+  target_compile_definitions(nvf_py_internal PRIVATE
+    "-DTORCH_CUDA_BUILD_MAIN_LIB"
+    "-DC10_BUILD_MAIN_LIB=1"
```

Comment:
I'm not confident here. I thought we are setting `C10_BUILD_MAIN_LIB` and ` TORCH_CUDA_BUILD_MAIN_LIB` because we are using c10/torch macro for export/import. But we might also need this since we are still linking some of their headers?

Reply:
Yeah, we are setting this because we are using some symbols from their headers. We haven't used their export macros for a while. And, something is seriously messed up in the whole c10 library situation. There's some `c10::Error` symbols (for example) that are undefined if we do not use these macros.

We should dig into this at some point, but as you can see this patch doesn't try to change the situation, and fixing every weird build sketchiness I encountered is an exercise in ocean boiling. I'll write up an issue, though.

Reply:
yeah torch packaging isn't bulletproof neither.